Sources/Testing/Traits/Tag.swift

```diff
@@ -10,56 +10,158 @@
+  ///
+  /// - Parameters:
+  ///   - stringValue: The user-supplied string value.
+  ///
+  /// Use this initializer when a user has provided an arbitrary string and it
+  /// is necessary to convert it into a tag. A simple heuristic is applied such
+  /// that the resulting instance may represent a (possibly non-existent) static
+  /// member of ``Tag`` or may represent a string literal.
+  @_spi(ForToolsIntegrationOnly)
+  public init(userProvidedStringValue stringValue: String) {
+    self.init(_codableStringValue: stringValue)
   }
+}
 
+// MARK: - CustomStringConvertible
+
+extension Tag: CustomStringConvertible {
   public var description: String {
-    rawValue
+    switch kind {
+    case let .staticMember(name):
+      ".\(name)"
+    case let .stringLiteral(stringLiteral):
+      #""\#(stringLiteral)""#
+    }
   }
 }
 
 // MARK: - Equatable, Hashable, Comparable
 
 extension Tag: Equatable, Hashable, Comparable {
-  public static func ==(lhs: Self, rhs: Self) -> Bool {
-    lhs.rawValue == rhs.rawValue
+  public static func <(lhs: Tag, rhs: Tag) -> Bool {
+    switch (lhs.kind, rhs.kind) {
+    case let (.staticMember(lhs), .staticMember(rhs)):
+      lhs < rhs
+    case let (.stringLiteral(lhs), .stringLiteral(rhs)):
+      lhs < rhs
+
+    // (Arbitrarily) sort static members before string literals.
+    case (.staticMember, .stringLiteral):
+      true
+    case (.stringLiteral, .staticMember):
+      false
+    }
   }
+}
 
-  public func hash(into hasher: inout Hasher) {
-    hasher.combine(rawValue)
+// MARK: - Codable, CodingKeyRepresentable
+
+extension Tag: Codable, CodingKeyRepresentable {
+  /// Initialize an instance of this type from a string previously encoded from
+  /// the `_codableStringValue` property.
+  ///
+  /// - Parameters:
+  ///   - stringValue: The previously-encoded string.
+  private init(_codableStringValue stringValue: String) {
+    if stringValue.first == "." {
+      self.init(kind: .staticMember(String(stringValue.dropFirst())))
+    } else if stringValue.first == #"\"# {
+      self.init(kind: .stringLiteral(String(stringValue.dropFirst())))
+    } else {
+      self.init(kind: .stringLiteral(stringValue))
+    }
   }
 
-  public static func <(lhs: Tag, rhs: Tag) -> Bool {
-    lhs.rawValue < rhs.rawValue
+  public init(from decoder: any Decoder) throws {
+    let stringValue = try String(from: decoder)
+    self.init(_codableStringValue: stringValue)
   }
-}
 
-// MARK: - Codable, CodingKeyRepresentable
+  /// This instance represented as a string, suitable for encoding.
+  private var _codableStringValue: String {
+    switch kind {
+    case let .staticMember(name):
+      ".\(name)"
+    case let .stringLiteral(stringLiteral):
+      if stringLiteral.first == "." || stringLiteral.first == #"\"# {
+        #"\\#(stringLiteral)"#
+      } else {
+        stringLiteral
+      }
+    }
+  }
+
+  public func encode(to encoder: any Encoder) throws {
+    try _codableStringValue.encode(to: encoder)
```

Comment:
This is encoding/decoding `Tag` as a single-value, right? I wonder if we may find that limiting, and would find it more future-proof to enclose it in a keyed container

Reply:
It is. If, in the future, we need more flexibility, we can adapt these methods to support multiple formats. Better for now to keep the encoding simple.

(Also, since we're currently using tags as keys in tag-colors.json, they need to be stringy, though that particular part of the library is easily forgotten.)